cmd/wharf/run.go

```diff
@@ -85,13 +91,23 @@ https://iver-wharf.github.io/#/usage-wharfyml/`,
 			return err
 		}
 
-		ctx := context.Background()
+		ctx, cancel := context.WithCancel(context.Background())
 		if runFlags.serve {
 			var server workerserver.Server
 			ctx, server = startWorkerServerWithCancel(ctx, store)
 			defer server.Close()
 		}
 
+		go handleCancelSignals(func() {
+			go func() {
+				time.AfterFunc(cancelGracePeriod, func() {
+					log.Warn().Message("Failed to cancel within grace period. Force quitting now.")
+					os.Exit(3)
+				})
+			}()
+			cancel()
+		})
```

Comment:
The code flow jumps back and forth a bunch with this callback method, and the logging is split up as well. Suggest to instead only pass in the `cancel()` function and let the `handleCancelSignals` do the logging and grace-period stuff, leaving the call to `handleCancelSignals` only be:

```go
go handleCancelSignals(cancel)
```